libs/datamodel/connectors/sql-datamodel-connector/src/postgres_datamodel_connector.rs

```diff
@@ -288,6 +290,71 @@ impl Connector for PostgresDatamodelConnector {
         Ok(())
     }
 
+    fn get_namespace_violations(&self, schema: &Datamodel) -> Vec<ConstraintNameSpace> {
+        let mut potential_name_space_violations: BTreeMap<(String, String), Vec<(String, String)>> = BTreeMap::new();
+
+        //Primary Key, Unique and Index names have to be globally unique
+        //Additionally, within a table they cannot conflict with Foreign Key names
+
+        for model in schema.models() {
+            if let Some(name) = model.primary_key.as_ref().and_then(|pk| pk.db_name.as_ref()) {
+                let entry = potential_name_space_violations
+                    .entry((name.to_string(), "pk, key, idx global".to_string()))
+                    .or_insert_with(Vec::new);
+
+                entry.push((model.name.clone(), "pk".to_string()));
+
+                let entry = potential_name_space_violations
+                    .entry((name.to_string(), format!("pk, key, idx, fk on {}", model.name)))
+                    .or_insert_with(Vec::new);
+
+                entry.push((model.name.clone(), "pk".to_string()));
+            }
+
+            for name in model
+                .relation_fields()
+                .filter_map(|rf| rf.relation_info.fk_name.as_ref())
+            {
+                let entry = potential_name_space_violations
+                    .entry((name.to_string(), format!("pk, key, idx, fk on {}", model.name)))
+                    .or_insert_with(Vec::new);
+
+                entry.push((model.name.clone(), "fk".to_string()));
+            }
+
+            for name in model.indices.iter().filter_map(|i| i.db_name.as_ref()) {
+                let entry = potential_name_space_violations
+                    .entry((name.to_string(), "pk, key, idx global".to_string()))
+                    .or_insert_with(Vec::new);
+
+                entry.push((model.name.clone(), "idx".to_string()));
```

Comment:
The `pk, idx. fk` naming in these methods is not super user friendly when it ends up in the error messages. It's also hard to understand for someone reading the code. Maybe an enum like

```rust
enum ConstraintNamespace {
  PrimaryKeys,
  ForeignKeys,
  Indexes,
  ...
}
```

with a `Display` impl would make the meaning clearer.

Reply:
Good idea, I reworked it.